geonode_mapstore_client/client/js/epics/gnsave.js

```diff
@@ -167,14 +179,16 @@ export const gnSaveDirectContent = (action$, store) =>
         .switchMap(() => {
             const state = store.getState();
             const mapInfo = mapInfoSelector(state);
-            const resourceId = mapInfo?.id // injected map id
+            const resourceId = mapInfo?.id
             || state?.gnresource?.id; // injected geostory id
             return Observable.defer(() => getResourceByPk(resourceId))
                 .switchMap((resource) => {
                     const metadata = {
-                        name: resource?.title,
-                        description: resource?.abstract,
-                        thumbnail: resource?.thumbnail_url
+                        name: (state.gnresource.data.name !== resource?.title ) ? state.gnresource.data.name : resource?.title,
```

Comment:
we colud use the selector or a constant instaed to reapeat state.gnresource.data to ensure state.gnresource.data exist

Reply:
ok, implement selectors